registeration/src/app/app.component.html

```diff
@@ -0,0 +1,79 @@
+<form class="form" [formGroup]="profileForm" (ngSubmit)="onSubmit()">
+  <div class="field">
+    <label class="label">userName</label>
+    <input
+      type="text"
+      class="input"
+      formControlName="userName"
+      [class.invalid]="
+        profileForm.get('userName')?.invalid &&
+        (profileForm.get('userName')?.touched ||
+          profileForm.get('userName')?.dirty)
+      "
+    />
+    <div
+      *ngIf="
+        profileForm.get('userName')?.invalid &&
+        (profileForm.get('userName')?.touched ||
+          profileForm.get('userName')?.dirty)
+      "
+      class="error-message"
+    >
+      User Name is required.
+    </div>
+  </div>
+
+  <div class="field">
+    <label class="label">email</label>
+    <input
+      type="email"
+      class="input"
+      formControlName="email"
+      [class.invalid]="
+        profileForm.get('email')?.invalid &&
```

Comment:
Create a function that checks all these conditions and returns only a boolean to simplify reading the html code